.github/workflows/windows.yml

```diff
@@ -0,0 +1,193 @@
+    runs-on: windows-2019-16-core
+    env:
+      CMAKE_BUILD_TYPE: 'Release'
+      CMAKE_CXX_COMPILER_LAUNCHER: ccache
+      CMAKE_C_COMPILER_LAUNCHER: ccache
+      OPENVINO_REPO: ${{ github.workspace }}\\openvino
+      OPENVINO_CONTRIB_REPO: ${{ github.workspace }}\\openvino_contrib
+      TEST_DATA: ${{ github.workspace }}\\testdata
+      INSTALL_DIR: ${{ github.workspace }}\\openvino\\install
+      BUILD_DIR: ${{ github.workspace }}\\openvino\\build
+      BUILD_DIR_CONTRIB: ${{ github.workspace }}\\openvino\\build_contrib
+      GRADLE_VER: '7.1.1'
+
+    steps:
+      - name: git configuration
+        run: git config --system core.longpaths true
+
+      - name: Clone OpenVINO
+        uses: actions/checkout@v4
+        with:
+          repository: 'openvinotoolkit/openvino'
+          path: ${{ env.OPENVINO_REPO }}
+          submodules: 'true'
+          ref: 'master'
+
+      - name: Clone OpenVINO Contrib
+        uses: actions/checkout@v4
+        with:
+          path: ${{ env.OPENVINO_CONTRIB_REPO }}
+          submodules: 'true'
+
+      - name: Clone Testdata
+        uses: actions/checkout@v4
+        with:
+          repository: 'openvinotoolkit/testdata'
+          path: ${{ env.TEST_DATA }}
+          lfs: 'true'
+          submodules: 'true'
+
+      #
+      # Dependencies
+      #
+
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+        with:
+          gradle-version: ${{ env.GRADLE_VER }}
+
+      - name: Setup Python ${{ env.PYTHON_VERSION }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+          cache: 'pip'
+
+      - name: Install python dependencies
+        run: |
+          # For Python API: build and wheel packaging
+          python3 -m pip install -r ${env:OPENVINO_REPO}/src/bindings/python/wheel/requirements-dev.txt
+
+      - name: Install build dependencies
+        run: |
+          Invoke-WebRequest https://github.com/ninja-build/ninja/releases/download/v1.11.1/ninja-win.zip -OutFile ninja-win.zip -MaximumRetryCount 10
+          Expand-Archive -Force ninja-win.zip
+          # Add it to the GitHub Path so it would be available in the subsequent steps
+          Add-Content -Path $env:GITHUB_PATH -Value "${{ github.workspace }}/ninja-win"
+          choco install opencv -y
+
+      #
+      # Build
+      #
+
+      - name: Configure Developer Command Prompt for Microsoft Visual C++
+        uses: ilammy/msvc-dev-cmd@v1
+
+      - name: Setup ccache
+        uses: hendrikmuhs/ccache-action@v1.2
+        with:
+          max-size: "2000M"
+          # Should save cache only if run in the master branch of the base repo
+          # github.ref_name is 'ref/PR_#' in case of the PR, and 'branch_name' when executed on push
+          save: ${{ github.ref_name == 'master' && 'true' || 'false'  }}
+          verbose: 2
+          key: ccache-windows
+          restore-keys: |
+            ccache-windows

```

Comment:
can we use sccache and share cache with main OV builds?

Reply:
As was discussed and suggested by @mryzhov and @dorloff, the GitHub-hosted runners will be used for the workflows in the Contrib repo and thus the GHA caches are used here.

Reply:
> GitHub-hosted runners

why?

Reply:
The main reasons as outline by @mryzhov and @dorloff are that the GH-hosted runners are from a different cost-center + `openvino_contrib` being more *open*. @mryzhov please correct/add if anything is incorrect.

Reply:
we agreed to use GHA runners and migrate on self-hosted runners later